plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -408,6 +417,125 @@ public void updatePartitionStatistics(HiveIdentity identity, Table table, String
         }
     }
 
+    @Override
+    public void updatePartitionStatistics(HiveIdentity identity, Table table, Map<String, Function<PartitionStatistics, PartitionStatistics>> updates)
+    {
+        Iterables.partition(updates.entrySet(), BATCH_CREATE_PARTITION_MAX_PAGE_SIZE).forEach(keySet -> {
+            updatePartitionStatisticsBatch(table, keySet);
+        });
+    }
+
+    private void updatePartitionStatisticsBatch(Table table, List<Map.Entry<String, Function<PartitionStatistics, PartitionStatistics>>> updates)
+    {
+        ImmutableList.Builder<BatchUpdatePartitionRequestEntry> updateRequests = ImmutableList.builder();
+        ImmutableSet.Builder<GlueColumnStatisticsProvider.StatisticsUpdate> statisticsUpdates = ImmutableSet.builder();
+
+        Map<String, UpdatePartitionAsyncHandler> asyncHandlers = updates.stream().collect(
+                toImmutableMap(
+                        Entry::getKey,
+                        entry -> new UpdatePartitionAsyncHandler(table, entry.getValue())));
+        Map<String, Future<GetPartitionResult>> getPartitionResults = updates.stream().collect(
+                toImmutableMap(
+                        Entry::getKey,
+                        entry -> glueClient.getPartitionAsync(new GetPartitionRequest()
```

Comment:
crap - I though one can use non-async `glueClient.getPartitions` here for simplicity. But it does not seem to allow to list a set of partition value lists in the request :(